rfd/0153-resource-guidelines.md

```diff
@@ -369,6 +374,82 @@ func (s *FooService) UpdateFoo(ctx context.Context, foo *foov1.Foo) (*foov1.Foo,
 }
 ```
 
+#### Patch
+
+For handling partial updates of a resource, the patch operation exists. The
+patch operation accepts the name of the resource and an update function. It 
+fetches the resource, calls the update function with the fetched resource and
+then calls `ConditionalUpdate` with the updated resource. It then returns the
+updated resource with the revision generated by the backend set.
+
+By default, the revision of the fetched resource is used for the
+`ConditionalUpdate`, however, it is possible for the mutation function to set
+the revision. This allows a patch operation to be performed conditionally
+against a specific revision.
```

Comment:
If the caller has a revision they probably have the whole resource, couldn't the caller just use `ConditionalUpdate`? What are the benefits of a conditional field mask-enabled update operation?

Reply:
This is a good point. Perhaps it's just easier to prohibit modifying the revision in Patch and suggest people use Update in those cases. It would definitely simplify the semantics of Patch.

Reply:
Ok great - agreed. I'll modify my RemoteCluster PR to reflect this.

Reply:
63de165c49

Reply:
Counterpoint: if the Get is part of a previous RPC, then using the client's view makes sense. Think `tctl edit` for example.

Reply:
I think given the case of `tctl edit`, it'd make more sense for `Update` to be used instead of `Patch` - the client already has a full view of the resource. `Patch` is more targetted towards something like `tctl update` which does not necessarily need a view of the full resource.

Reply:
`tctl edit` can still race against someone else's edit, potentially overwriting changes.

Reply:
`tctl edit` is already using conditional update semantics for the resource types for which it's possible to do so.